### PR TITLE
[Merged by Bors] - chore(CategoryTheory/Sites): make `Subcanonical` a class

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dagurtomas** [comment]
!bench

**Comment by joelriou** [comment]
> Here are the [benchmark results](http://speed.lean-fro.org/mathlib4/run-detail/0ce3f622-3abf-4ba2-b54f-13ec155774fc) for commit [4693fc8](https://github.com/leanprover-community/mathlib4/commit/4693fc884aa51b8eb31fd8c1aa70bacb726aca73). There were [significant changes](http://speed.lean-fro.org/mathlib4/compare/c6110939-c614-4eb3-90ac-2ec30fb53853/to/0ce3f622-3abf-4ba2-b54f-13ec155774fc) against commit [f7b57bc](https://github.com/leanprover-community/mathlib4/commit/f7b57bce1a08771bc6b7d35d4275bb6aba0697b5):
> 
> ```diff
>   Benchmark      Metric                 Change
>   =====================================================
> - build          .olean serialization    11.2%
> - build          import                  11.6%
> - build          initialization           9.6%
> - open Mathlib   task-clock              18.1% (78.4 σ)
> ```

I am completely confused about what this means!

**Comment by dagurtomas** [comment]
> > Here are the [benchmark results](http://speed.lean-fro.org/mathlib4/run-detail/0ce3f622-3abf-4ba2-b54f-13ec155774fc) for commit [4693fc8](https://github.com/leanprover-community/mathlib4/commit/4693fc884aa51b8eb31fd8c1aa70bacb726aca73). There were [significant changes](http://speed.lean-fro.org/mathlib4/compare/c6110939-c614-4eb3-90ac-2ec30fb53853/to/0ce3f622-3abf-4ba2-b54f-13ec155774fc) against commit [f7b57bc](https://github.com/leanprover-community/mathlib4/commit/f7b57bce1a08771bc6b7d35d4275bb6aba0697b5):
> > ```diff
> >   Benchmark      Metric                 Change
> >   =====================================================
> > - build          .olean serialization    11.2%
> > - build          import                  11.6%
> > - build          initialization           9.6%
> > - open Mathlib   task-clock              18.1% (78.4 σ)
> > ```
> 
> I am completely confused about what this means!

So am I... I'll try again

File: Mathlib/CategoryTheory/Sites/Canonical.lean [hunk 1]
```diff
@@ -205,60 +205,83 @@ theorem isSheaf_of_isRepresentable (P : Cᵒᵖ ⥤ Type v) [P.IsRepresentable]
...
+lemma le_canonical (J : GrothendieckTopology C) [Subcanonical J] : J ≤ canonicalTopology C :=
+  Subcanonical.le_canonical
+
+instance : (canonicalTopology C).Subcanonical where
+  le_canonical := le_rfl
 
 namespace Subcanonical
 
 /-- If every functor `yoneda.obj X` is a `J`-sheaf, then `J` is subcanonical. -/
 theorem of_yoneda_isSheaf (J : GrothendieckTopology C)
```
[the remark this answers is not in the document]
Comment (dagurtomas) [reply Mathlib/CategoryTheory/Sites/Canonical.lean]: I renamed it to `of_isSheaf_yoneda_obj`, which seems to be the most correct

**Comment by dagurtomas** [comment]
!bench

**Comment by dagurtomas** [comment]
!bench

**Comment by dagurtomas** [comment]
Ok, merging master got rid of these weird benchmark results.

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**